test/fit_tests/deploy/onrack_stack_init.py

```diff
@@ -0,0 +1,306 @@
+                         + str(api_data['status']))
+
+    # Optionally install PDU node if present
+    @fit_common.unittest.skipUnless("pdu" in fit_common.STACK_CONFIG[fit_common.ARGS_LIST['stack']], "")
+    def test05_discover_pdu_node(self):
+        print "**** Creating PDU node."
+        payload = {
+                    "type":"pdu",
+                    "name":"PDU",
+                    "autoDiscover":True,
+                    "snmpSettings":{
+                        "host": fit_common.STACK_CONFIG[fit_common.ARGS_LIST['stack']]['pdu'],
+                        "community": fit_common.GLOBAL_CONFIG['snmp']['community'],
+                    }
+                    }
+        api_data = fit_common.rackhdapi("/api/2.0/nodes", action='post', payload=payload)
+        self.assertEqual(api_data['status'], 201, 'Incorrect HTTP return code, expecting 201, got '
+                         + str(api_data['status']))
+
+    def test06_check_compute_nodes(self):
+        print "**** Waiting for compute nodes."
+        c_index = 0
+        for c_index in range(0, MAX_CYCLES):
+            if "compute" in fit_common.rackhdapi("/api/2.0/nodes")['text']:
+                break
+            else:
+                fit_common.time.sleep(30)
+        self.assertLess(c_index, MAX_CYCLES-1, "No compute nodes settings found.")
+
+    def test07_check_discovery(self):
+        print "**** Waiting for node Discovery to complete.\n",
+        # Determine if there are any active workflows. If returned value is true, obmSettings, SKUs
+        # and active workflows are all either present or complete. If  the returned is false,
+        # there was a timeout and all nodes have not obtained obmSetting, SKUs, or all active
+        # workflows have not completed.
+        # Wait 10 minutes ( MAX_CYCLES * 10 seconds) for this to occur.
+        self.assertTrue(self.check_for_active_workflows(MAX_CYCLES), "Node discovery not completed")
+        
+    def check_for_active_workflows(self, max_time):
+        '''
+        Determine if are any active workflows.
+        :param    Time to wait (in 10 second intervals)
+        :return:  True  - No active workflows
+                  False - Workflows are active
+        '''
+        for dummy in range(0, max_time):
+            nodes_data = fit_common.rackhdapi("/api/2.0/nodes")
+            if nodes_data['status'] == 200 and len(nodes_data['json']) > 0:
+                # if there are nodes present, determine if discovery has completed on them
+                discovery_complete = True
+
+                for node in nodes_data['json']:
+                    if node['type'] == 'compute':
+                        self.assertIn('id', node, 'node does not contain id')
+                        node_id = node['id']
+                        # determine if there are any active worlflows. If so, discovery not completed
+                        awf_data = fit_common.rackhdapi('/api/1.1/nodes/'
```

Comment:
Agreed that we should migrate  to 2.0 but while this was being coded 2.0 APIs where changing rapidly so used 1.1 as a stable base. Migrating all base code to 2.0 is planned.